core/src/main/java/com/scalar/db/storage/cosmos/SelectStatementHandler.java

```diff
@@ -135,6 +147,46 @@ private String makeQueryWithCondition(
     return select.getSQL(ParamType.INLINED);
   }
 
+  private SelectJoinStep<org.jooq.Record> makeQueryWithProjections(
+      Selection selection, TableMetadata tableMetadata) {
+    if (selection.getProjections().isEmpty()) {
+      return DSL.using(SQLDialect.DEFAULT).select().from("Record r");
+    }
+
+    List<String> projectedFields = new ArrayList<>();
+
+    // To project the required columns, we build a JSON object with the same structure as the
+    // `Record.class`so that each field can be deserialized properly into a `Record.class` object.
+    // For example, the projected field "r.id" will be mapped to the `Record.id` attribute
+    projectedFields.add("r.id");
+    projectedFields.add("r.concatenatedPartitionKey");
+    projectedFields.add("r.partitionKey");
+    projectedFields.add("r.clusteringKey");
+
+    List<String> projectedValues =
+        selection.getProjections().stream()
+            .filter(
+                c ->
+                    !tableMetadata.getPartitionKeyNames().contains(c)
+                        && !tableMetadata.getClusteringKeyNames().contains(c))
+            .map(c -> "\"" + c + "\":r.values" + CosmosUtils.quoteKeyword(c))
+            .collect(Collectors.toList());
+
+    if (!projectedValues.isEmpty()) {
+      // The following will be mapped to the "Record.values" map attribute
+      // For example, to project the columns c1 and c2, the values field will be
+      // `{"c1": r["c1"], "c2":r["c2"]} as values`
+      // Besides, since the Jooq parser consumes curly brace character as they are treated as
+      // placeholder, each curly brace need to be doubled "{{" to have a single curly brace "{"
+      // present in the generated sql query
+      projectedFields.add("{{" + String.join(",", projectedValues) + "}} as values");
+    }
+
+    return DSL.using(SQLDialect.DEFAULT)
+        .select(projectedFields.stream().map(DSL::field).collect(Collectors.toList()))
+        .from("Record r");
+  }
+
```

Comment:
The query projections are handled here. 
For example, to project the columns `c3` and `c4`, the request for a ScanAll will be like : 
`select r.id, 
             r.concatenatedPartitionKey, 
             r.partitionKey, 
             r.clusteringKey, 
             {"name3":r.values["name3"],"name4":r.values["name4"]} as values 
             from Record r`